wandb/sdk/wandb_run.py

```
@@ -3112,52 +3112,98 @@ def _prepare_artifact(
+            name: (str) A model artifact name. 'name' must match the name of an existing logged
+                model artifact.
                 May be prefixed with entity/project/. Valid names
                 can be in the following forms:
-                    - name:version
-                    - name:alias
-                    - digest.
+                    - model_artifact_name:version
+                    - model_artifact_name:alias
+                    - model_artifact_name:digest.
+
+        Examples:
+            ```python
+            run.use_model(
+                name="my_model_artifact:latest",
+            )
+
+            run.use_model(
+                name="my_project/my_model_artifact:v0",
+            )
+
+            run.use_model(
+                name="my_entity/my_project/my_model_artifact:<digest>",
+            )
+            ```
+
+            Invalid usage
+            ```python
+            run.use_model(
+                name="my_entity/my_project/my_model_artifact",
+            )
+            ```
 
         Raises:
-            AssertionError: if type of artifact 'model_name' does not contain 'model'
+            AssertionError: if model artifact 'name' is of a type that does not contain the substring 'model'.
+
         Returns:
-            path: (StrPath) path to downloaded artifact file(s).
+            path: (str) path to downloaded model artifact file(s).
         """
-        artifact = self.use_artifact(artifact_or_name=model_name)
+        artifact = self.use_artifact(artifact_or_name=name)
         assert "model" in str(
             artifact.type.lower()
-        ), "You can only use this method for 'model' artifacts. Please make sure the artifact type of the model you're trying to use contains the word 'model'."
+        ), "You can only use this method for 'model' artifacts. For an artifact to be a 'model' artifact, its type property must contain the substring 'model'."
```

do you want to raise an assertion exception here?

---

Nope! This assertion is to make sure users only link artifacts of type 'model', similar to how we handle it in `use_model()`

---

Nope! Nope! This assertion is to make sure users use this method only to retrieve their `model` files. For anything else, they can use `use_artifact()`